src/dsp_tools/utils/rosetta.py

```diff
@@ -0,0 +1,67 @@
+from pathlib import Path
+import shutil
+import subprocess
+
+from dsp_tools.models.exceptions import UserError
+from dsp_tools.utils.project_create import create_project
+from dsp_tools.utils.xml_upload import xml_upload
+
+def upload_rosetta() -> bool:
+    """
+    This method clones https://github.com/dasch-swiss/082E-rosetta-scripts
+    into ~/.dsp-tools/rosetta.
+    If the repository is already there, it pulls instead of cloning.
+    Then, rosetta.json is created and rosetta.xml uploaded.
+
+    Raises:
+        UserError: If the repo cannot be cloned nor pulled
+
+    Returns:
+        True if everything went well
+    """
+
+    enclosing_folder = Path.home() / Path(".dsp-tools/rosetta")
+    enclosing_folder.mkdir(parents=True, exist_ok=True)
+    rosetta_folder = enclosing_folder / "082E-rosetta-scripts"
+    
+    if rosetta_folder.is_dir():
+        cloning_necessary = False
+        print(f"Make 'git pull' in {rosetta_folder}...")
+        completed_process = subprocess.run("git pull", shell=True, cwd=rosetta_folder)
+        if not completed_process or completed_process.returncode != 0:
+            print(f"'git pull' failed. Remove '{rosetta_folder}'...")
+            shutil.rmtree(rosetta_folder, ignore_errors=True)
+            cloning_necessary = True
+    else:
+        cloning_necessary = True
+    
+    if cloning_necessary:
```

Comment:
the `cloning_necessary` is redundant. You can just do `if rosetta_folder.is_dir(): git-pull; else: git-clone`

Reply:
Strictly speaking, it is not redundant: This mechanism allows that if `git pull` fails, it falls back to cloning the repo again. This adds more robustness (e.g. if the old clone is damaged). 
But from the perspective of requirements analysis, we can of course discuss if this mechanism is necessary.